aslam_offline_calibration/kalibr/python/kalibr_common/ImageDatasetReader.py

```diff
@@ -134,7 +134,7 @@ def getImage(self, idx):
       img_data = (image_16u / 256).astype("uint8")
     elif data.encoding == "8UC1" or data.encoding == "mono8":
       img_data = np.array(self.CVB.imgmsg_to_cv2(data))
-    elif data.encoding == "8UC3" or data.encoding == "rgb8":
+    elif data.encoding == "8UC3" or data.encoding == "rgb8" or data.encoding == "bgr8":
       img_data = np.array(self.CVB.imgmsg_to_cv2(data))
       img_data = cv2.cvtColor(img_data, cv2.COLOR_BGR2GRAY)
```

Comment:
These two lines look a bit fishy to me. `self.CVB.imgmsg_to_cv2(data)` returns a cv image with the same encoding as the input image, which is either `8UC3` (unknown color channels), `rgb8` or `bgr8`. All of them are then interpreted as BGR in the next line with `cv2.COLOR_BGR2GRAY`.

It looks like the different channels get different weights when converting to grayscale (https://docs.opencv.org/3.1.0/de/d25/imgproc_color_conversions.html), so rgb and bgr should be distinguished in this if-then-else block. Not sure what order to assume for 8UC3 though...

PS: This is of course not an issue introduced by your PR. I'm not a maintainer, so I have no say in whether this should be fixed here or left as is.

Reply:
Both suggestions sound reasonable to me. A maintainer will have to decide though, I guess.